regular_expressions/regular_expressions.md

```diff
@@ -0,0 +1,623 @@
+
+**Other elements**
+
+You might notice a language selector on the left indicating what "flavor" of regular expressions to use. Generally, we suggest PCRE2 to learn from, as this flavor (Perl Compatible Regular Expressions 2) is widely adopted across languages and any differences between PCRE2 and the flavor used in your environment is likely to be small.
+
+### An Aside: Flags
+
+The regular expression checker we refer to and use for this module, [Regex 101](https://www.regex101.com), has, by default, the "global" and "multiline" flags set.  We won't get into setting flags, which is something you will learn when you start using regular expressions within the context of a programming language.
+
+If you do some internet searches or use a regular expression checker, you might wondering why regular expressions are often put between forward slashes followed by a lower case `g`, or lower case `gm`, it's because of how regular expressions are written in some programming languages like Perl and Javascript and some utilities like `sed`.  
+
+![](media/example1.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width: 400px;" -->
+
+![](media/example2.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width: 300px;" -->
+
+![](media/example3.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width: 400px;" -->
+
+
+You don't need to worry about enclosing your regular expressions in forward slashes for this module!  However, if you do use different "flags" than we do, you might see an occasional difference in how the regular expression we show works in your environment.  Try setting your flags to "global" and "multiline", and you should get identical results to what we show here.
+
+## Simple Regular Expressions
+
+To describe a pattern, you need to indicate what characters you expect to see, and how many of them, for each section of the pattern.  Most typically, we work left to right, and describe a pattern in order.  
+
+We'll start out by describing the two most important elements of regex: describing which characters, and then quantifying how many of those characters you expect to find at that point in the pattern.
+
+Soon, the mysterious set of symbols that makes up a regular expression will make sense and not feel like random characters!
+
+
+### Which Characters?
+
+Patterns defined in regex often involve special groups of [unicode characters](https://en.wikipedia.org/wiki/List_of_Unicode_characters).  For example, alphabetic characters (a-z and A-Z) are made of several groups, digits (0-9) form a group, whitespace characters like tab and space are another group of characters.  Maybe your patient identifiers begin with three letters (a special group), then have a hyphen (only a single character is valid here), then have four digits (another special group).  You can use predefined groups or define your own.
+
+Here are some common ways to designate which characters form part of a pattern:
```

Comment:
Maybe break this giant list into a few subsections? 

I suggest start with matching exact letters and any digits, then any letters, then add things on a few at a time. This whole list could be great as a reference at the end of the section/module.